flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/NoFetchingInput.java

```diff
@@ -114,18 +117,26 @@ public void readBytes(byte[] bytes, int offset, int count) throws KryoException
             throw new IllegalArgumentException("bytes cannot be null.");
         }
 
+        if (count == 0) {
+            return;
+        }
+
```

Comment:
This `if (count == 0) {` is alternative change for FLINK-34954, I ran the demo that provided in FLINK-34954, it works well. So I think the change is fine.

Also, I ran the benchmark, the performance of serializerKryo and serializerKryoWithoutRegistration are recovered.

Reply:
Hi @dannycranmer , this issue is the blocker of Flink 1.20, and you are the reviewer of https://github.com/apache/flink/pull/24586, would you mind reviewing this PR as well? thanks in advance.

Reply:
@1996fanrui Can you add the demo from FLINK-34954 as a test?

Reply:
Thanks @uce for the reminder!

Let's follow it up at https://github.com/apache/flink/pull/24837 .